### ISSUE TITLE
midpoint error edge.length is not present

### ISSUE THREAD

**Issue by llrs** [KlausVigo/phangorn#38]
midpoint function `node2root` expects an element `edge.length`
https://github.com/KlausVigo/phangorn/blob/a141097fad2fa89d81fdd4698ba509a24bca9494/R/treeManipulation.R#L142
However in some trees it is not present, and this results in an error:

```
tree <- read.tree("input_file.nwk")
midpoint(tree)
Error in el[child] <- x$edge.length : replacement has length zero
```
I can provide the file I have problems with if required. 

```
R version 3.4.2 (2017-09-28)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows XP (build 2600) Service Pack 3

Matrix products: default

locale:
  [1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
  [1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
  [1] fpc_2.1-10      cluster_2.0.6   phangorn_2.2.0  GUniFrac_1.0   
[5] ape_5.0         vegan_2.4-4     lattice_0.20-35 permute_0.9-4  
[9] ade4_1.7-10    

loaded via a namespace (and not attached):
  [1] igraph_1.1.2      flexmix_2.3-14    Rcpp_0.12.13      magrittr_1.5     
[5] MASS_7.3-47       mclust_5.3        quadprog_1.5-5    fastmatch_1.1-0  
[9] prabclus_2.2-6    tools_3.4.2       nnet_7.3-12       parallel_3.4.2   
[13] grid_3.4.2        nlme_3.1-131      mgcv_1.8-22       class_7.3-14     
[17] modeltools_0.2-21 Matrix_1.2-11     kernlab_0.9-25    trimcluster_0.1-2
[21] robustbase_0.92-7 DEoptimR_1.0-8    compiler_3.4.2    diptest_0.75-7   
[25] stats4_3.4.2      mvtnorm_1.0-6     pkgconfig_2.0.1  
```

**Comment by KlausVigo** [comment]
Dear @llrs,
thanks for spotting this. I think fixing this is a bit tricky and one would expect a tree to have edge length if one wants to apply midpoint rooting. Otherwise one would need to add arbitrarily some edge length, e.g. set all to 1 or draw from a distribution (e.g. `rexp`) or set all edges to zero and root randomly. 
Maybe the cleanest solution is to issue a warning and return the tree as it is (or NULL), as this would also work for `multiPhylo` objects.

**Comment by llrs** [comment]
I think issuing a warning and returning the tree as is will work for me.
I will review how I created the tree as it worked in other cases and I think I did the same in that case.

**Comment by KlausVigo** [comment]
commit 38db073010be1910caeee8b1447257674c365d25 closes #38
